airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftSqlGenerator.java

```diff
@@ -0,0 +1,543 @@
+import io.airbyte.integrations.base.destination.typing_deduping.Array;
+import io.airbyte.integrations.base.destination.typing_deduping.ColumnId;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamConfig;
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId;
+import io.airbyte.integrations.base.destination.typing_deduping.Struct;
+import io.airbyte.integrations.base.destination.typing_deduping.Union;
+import io.airbyte.integrations.base.destination.typing_deduping.UnsupportedOneOf;
+import io.airbyte.protocol.models.v0.DestinationSyncMode;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.jooq.CommonTableExpression;
+import org.jooq.Condition;
+import org.jooq.CreateSchemaFinalStep;
+import org.jooq.CreateTableColumnStep;
+import org.jooq.DSLContext;
+import org.jooq.DataType;
+import org.jooq.Field;
+import org.jooq.InsertValuesStepN;
+import org.jooq.Name;
+import org.jooq.Record;
+import org.jooq.SQLDialect;
+import org.jooq.SelectConditionStep;
+import org.jooq.conf.ParamType;
+import org.jooq.impl.DSL;
+import org.jooq.impl.DefaultDataType;
+import org.jooq.impl.SQLDataType;
+
+public class RedshiftSqlGenerator extends JdbcSqlGenerator {
+
+  public static final String CASE_STATEMENT_SQL_TEMPLATE = "CASE WHEN {0} THEN {1} ELSE {2} END ";
+  public static final String CASE_STATEMENT_NO_ELSE_SQL_TEMPLATE = "CASE WHEN {0} THEN {1} END ";
+  private static final Map<String, String> REDSHIFT_TYPE_NAME_TO_JDBC_TYPE = ImmutableMap.of(
+      "float8", "float",
+      "int8", "bigint",
+      "bool", "boolean",
+      "timestamptz", "timestamp with time zone",
+      "timetz", "time with time zone");
+  private static final String COLUMN_ERROR_MESSAGE_FORMAT = "Problem with `%s`";
+  private static final String AIRBYTE_META_COLUMN_ERRORS_KEY = "errors";
+
+  private final ColumnId CDC_DELETED_AT_COLUMN = buildColumnId("_ab_cdc_deleted_at");
+
+  public RedshiftSqlGenerator(final NamingConventionTransformer namingTransformer) {
+    super(namingTransformer);
+  }
+
+  /**
+   * This method returns Jooq internal DataType, Ideally we need to implement DataType interface with
+   * all the required fields for Jooq typed query construction
+   *
+   * @return
+   */
+  private DataType<?> getSuperType() {
+    return new DefaultDataType<>(null, String.class, "super");
+  }
+
+  @Override
+  protected DataType<?> getStructType() {
+    return getSuperType();
+  }
+
+  @Override
+  protected DataType<?> getArrayType() {
+    return getSuperType();
+  }
+
+  @Override
+  protected DataType<?> getWidestType() {
+    return getSuperType();
+  }
+
+  @Override
+  protected SQLDialect getDialect() {
+    return SQLDialect.POSTGRES;
+  }
+
+  protected DSLContext getDslContext() {
+    return DSL.using(getDialect());
```

Comment:
Default is unformatted SQL sent over the wire. For tests I overrode this method to print formatted SQL. Should we keep this formatted and incur additional byte overhead but make debuggin easy in Job logs of the sync ?

Reply:
tbh I think unformatted is better. It looks like this:
![image](https://github.com/airbytehq/airbyte/assets/5741425/1292d3a5-872d-4b91-96cc-3df2f5b8fc6c)

which... seems _more_ readable, since we're not blasting 1000 lines of pretty-printed sql into the logs

anyone who needs to read the sql is going to copy-paste it into a text editor anyway.